core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -337,7 +337,18 @@ class ZkPartitionStateMachine(config: KafkaConfig,
 
       finished.foreach {
         case (partition, Left(e)) =>
-          logFailedStateChange(partition, partitionState(partition), OnlinePartition, e)
+          // Extract failure message, if present, to append to error log.
+          val reasonMsg = if (e.isInstanceOf[StateChangeFailedException]) {
+            s", reason: ${e.getMessage}"
+          } else {
+            ""
+          }
+          val failMsg = s"Controller failed to change state for partition $partition from ${partitionState(partition)} to $OnlinePartition"
```

Comment:
nit: since the log identity already starts with "Controller," I think we can leave it out and start from "Failed."

Reply:
Done